### PR TITLE
MM-54182 - add correct information to session table from mobile devices

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR modifies the doLogin function to correctly capture the user agent information so mobile and desktop app are identified correctly and these information is stored correctly in the sessions table.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54182

#### Screenshots

https://github.com/mattermost/mattermost/assets/10082627/131059b3-5709-4eb5-b7e8-394310791024

Before:

After:

https://github.com/mattermost/mattermost/assets/10082627/2ca94898-2ee6-4572-89c5-9a7bc6b4940f

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by BenCookie95 — APPROVED** [review]
LGTM! Do we need a review from someone on the mobile team?

**Comment by pvev** [comment]
@JulienTant friendly reminder on this one :)

**Review by larkox — APPROVED** [review]
LGTM. The only nit I would say is that maybe we should be more clear in the code that this only checks for mobile app. A mobile web browser won't be distinguishable from any other browser session.

**Comment by pvev** [comment]
> LGTM. The only nit I would say is that maybe we should be more clear in the code that this only checks for mobile app. A mobile web browser won't be distinguishable from any other browser session.

thanks @larkox! I will add the comment there to make it clear which mobile we are referring to.

**Comment by lindy65** [comment]
Hi @pvev - please add QA test steps to the Jira ticket?  Thanks!

**Comment by pvev** [comment]
> Hi @pvev - please add QA test steps to the Jira ticket? Thanks!

hi @lindy65 sure!

File: server/channels/utils/utils_test.go [hunk 1]
```diff
@@ -279,6 +280,32 @@ func TestRoundOffToZeroes(t *testing.T) {
...
+	testCases := []struct {
+		userAgent string
+		expected  bool
+	}{
+		// Test cases with mobile devices
+		{"Mozilla/5.0 (iPhone; CPU iPhone OS 15_0 like Mac OS X)", true},
+		{"Mozilla/5.0 (Android 12; Mobile)", true},
+		{"Mozilla/5.0 (Linux; Android 12; Pixel 6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.9999.99 Mobile Safari/537.36", true},
+
+		// Test cases with NO movile devices
```
Comment (JulienTant) [inline server/channels/utils/utils_test.go:293]: nit
```suggestion
		// Test cases with NO mobile devices
```

**Review by JulienTant — APPROVED** [review]
LGTM Pablo!

File: server/channels/utils/utils_test.go (hunk 1 above)
Comment (pvev) [reply server/channels/utils/utils_test.go:293]: good catch! that's my Spanglish :)

**Review by lindy65 — APPROVED** [review]
Thanks @pvev - tested as per steps in the Jira ticket and it looks good to me 👍

**Pull request merged**